src/awesome_map/BoundaryInterceptor.js

```diff
@@ -225,6 +243,37 @@ define(function(require) {
             var eventType = event.type;
             var originalState;
 
+
+            // Boundary Detection
+            // @TODO: It seems that this event can be triggered by inertial events generated
+            //  by the browser.  Is this desirable or no?  Is there any way to prevent this?
+            var boundedPosition = this._getBoundedPosition(targetState);
+            var currentState = this._awesomeMap.getCurrentTransformState();
+            if (currentState.translateY === boundedPosition.y) {
+                if (targetState.translateY-boundedPosition.y > this._boundarySensitivity) {
+                    this.onScrollPastBoundary.dispatch([this, {
+                        boundary: BoundaryTypes.TOP
+                    }]);
+                }
+                if (targetState.translateY-boundedPosition.y < -this._boundarySensitivity) {
+                    this.onScrollPastBoundary.dispatch([this, {
+                        boundary: BoundaryTypes.BOTTOM
+                    }]);
+                }
+            }
+            if (currentState.translateX === boundedPosition.x) {
```

Comment:
It might be interesting for the plustenner to test this with a jagged doc; one with pages of varying widths.